scripting/src/main/java/org/infinispan/scripting/impl/LocalRunner.java

```diff
@@ -0,0 +1,30 @@
+package org.infinispan.scripting.impl;
+
+import java.util.concurrent.Future;
+
+import javax.script.CompiledScript;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.infinispan.commons.util.concurrent.NoOpFuture;
+
+public class LocalRunner implements ScriptRunner {
+   public static final LocalRunner INSTANCE = new LocalRunner();
+
+   private LocalRunner() {
+   }
+
+   @Override
+   public <T> Future<T> runScript(ScriptingManagerImpl scriptManager, String scriptName, ScriptMetadata metadata, CacheScriptBindings bindings) throws ScriptException {
+      CompiledScript compiled = scriptManager.compiledScripts.get(scriptName);
+      if (compiled != null) {
+         T result = (T) compiled.eval(bindings);
+         return new NoOpFuture<T>(result);
+      } else {
+         ScriptEngine engine = scriptManager.getEngineForScript(scriptName, metadata);
+         T result = (T) engine.eval(scriptManager.scriptCache.get(scriptName), bindings);
```

Comment:
I'm wondering. Could someone with EXEC permission write a malicious script and cause real havoc in the cluster? If so, have you considered any form of sandboxing to prevent arbitrary code execution, something like [1], but that would cover all engines? 

[1] http://groovy-sandbox.kohsuke.org/